Various changes to airlift http client/server

nezihyigitbasi opened the pull request:
Depends on #587.

There are two main changes:
- Disable remote async error notifications in HttpServer. This is a workaround for https://github.com/jersey/jersey/issues/3691 for HTTP/2
- Add HTTP/2 stream idle timeout config to HTTP server.

nezihyigitbasi commented:
/cc @electrum

electrum commented on http-server/src/main/java/io/airlift/http/server/HttpServer.java:
```diff
@@ -127,16 +127,17 @@ public HttpServer(HttpServerInfo httpServerInfo,
+        baseHttpConfiguration.setSendServerVersion(false);
+        baseHttpConfiguration.setSendXPoweredBy(false);
+        if (config.getMaxRequestHeaderSize() != null) {
+            baseHttpConfiguration.setRequestHeaderSize(Ints.checkedCast(config.getMaxRequestHeaderSize().toBytes()));
```
While you're changing this, update it to `toIntExact()`

electrum commented on http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java:
```diff
@@ -111,6 +112,7 @@ public void testExplicitPropertyMappings()
                 .put("http-server.http2.session-receive-window-size", "4MB")
                 .put("http-server.http2.stream-receive-window-size", "4MB")
                 .put("http-server.http2.input-buffer-size", "4MB")
+                .put("http-server.http2.stream-idle-timeout", "20s")
```
It's usually better to use a non-round value (random-ish) like `23s` that wouldn't ever be used as a default.

electrum commented on http-client/src/main/java/io/airlift/http/client/jetty/JettyIoPool.java:
```diff
@@ -130,7 +130,7 @@ public String toString()
         private final ScheduledExecutorService[] schedulers;
         private final ThreadFactory threadFactory;
 
-        public ConcurrentScheduler(int schedulerCount, int threadsPerScheduler, String threadBaseName)
```
Nit: I'd title the commit "Minor cleanup" instead of "refactor"

nezihyigitbasi commented:
Addressed your comments and added a new commit (7cb61e0) to fix the `AbstractHttpClientTest::testConnectTimeout()` test

```
[ERROR] testConnectTimeout(io.airlift.http.client.jetty.TestAsyncJettyHttpClientHttp2)  Time elapsed: 0.031 s  <<< FAILURE!
java.lang.AssertionError: 
unexpected exception: [java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.http2.HTTP2Session.onShutdown(HTTP2Session.java:878)
	at org.eclipse.jetty.http2.HTTP2Connection$HTTP2Producer.produce(HTTP2Connection.java:231)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:179)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.eclipse.jetty.http2.HTTP2Connection.onOpen(HTTP2Connection.java:97)
	at org.eclipse.jetty.http2.client.HTTP2ClientConnectionFactory$HTTP2ClientConnection.onOpen(HTTP2ClientConnectionFactory.java:148)
	at org.eclipse.jetty.io.SelectorManager.connectionOpened(SelectorManager.java:322)
	at org.eclipse.jetty.io.ManagedSelector.createEndPoint(ManagedSelector.java:213)
	at org.eclipse.jetty.io.ManagedSelector.access$1600(ManagedSelector.java:60)
	at org.eclipse.jetty.io.ManagedSelector$CreateEndPoint.run(ManagedSelector.java:650)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:748)
```

electrum commented:
Merged, thanks!
